File: benchmarks/reference/c/fib_stress.c
/* fib_stress.c — Recursive fibonacci(35) benchmark.
   Equivalent to examples/il/benchmarks/fib_stress.il */
#include <stdint.h>
#include <stdlib.h>

static int64_t fib(int64_t n)
{
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
}

int main(void)
{
    int64_t result = fib(35);
    return (int)(result & 0xFF);
}

File: benchmarks/reference/rust/fib_stress.rs
fn fib(n: i64) -> i64 {
    if n <= 1 {
        n
    } else {
        fib(n - 1) + fib(n - 2)
    }
}

fn main() {
    let args_offset = std::env::args().count() as i64 - 1;
    let result = fib(40 + args_offset);
    std::process::exit((result & 0xFF) as i32);
}
